### PR TITLE
openmeeg 2.4.prerelease

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

**Comment by iMichka** [comment]
Fails on mac with:
> /tmp/openmeeg-20170512-4674-5237wh/openmeeg-release-2.1/contrib/matio/src/mat73.c:822:62: error: too few arguments to function call, expected 4, have 3
                                                    ref_ids+l);
                                                             ^
/usr/local/include/H5Rpublic.h:61:8: note: 'H5Rdereference2' declared here
H5_DLL hid_t H5Rdereference2(hid_t obj_id, hid_t oapl_id, H5R_type_t ref_type, const void *ref);
       ^
/tmp/openmeeg-20170512-4674-5237wh/openmeeg-release-2.1/contrib/matio/src/mat73.c:971:72: error: too few arguments to function call, expected 4, have 3
                    refid = H5Rdereference(dset_id,H5R_OBJECT,ref_ids+i);
                            ~~~~~~~~~~~~~~                             ^
/usr/local/include/H5Rpublic.h:61:8: note: 'H5Rdereference2' declared here
H5_DLL hid_t H5Rdereference2(hid_t obj_id, hid_t oapl_id, H5R_type_t ref_type, const void *ref);
       ^
/tmp/openmeeg-20170512-4674-5237wh/openmeeg-release-2.1/contrib/matio/src/mat73.c:2621:77: error: too few arguments to function call, expected 4, have 3
                        ref_id = H5Rdereference(dset_id,H5R_OBJECT,ref_ids+i);
                                 ~~~~~~~~~~~~~~                             ^
/usr/local/include/H5Rpublic.h:61:8: note: 'H5Rdereference2' declared here
H5_DLL hid_t H5Rdereference2(hid_t obj_id, hid_t oapl_id, H5R_type_t ref_type, const void *ref);
       ^
3 errors generated.

**Comment by iMichka** [comment]
Reported upstream: https://github.com/openmeeg/openmeeg/issues/203

File: openmeeg.rb [hunk 1]
```diff
@@ -1,15 +1,28 @@
...
+  desc "Package for low-frequency bio-electromagnetism solving"
+  homepage "https://openmeeg.github.io"
+  url "https://github.com/openmeeg/openmeeg/archive/2.4.prerelease.tar.gz"
+  version "2.4.prerelease"
+  sha256 "fabf9addfcb99d3b3161acde98104a321ec926a1968e444edfce5334adb1ae87"
 
   head "https://github.com/openmeeg/openmeeg.git"
 
   depends_on "cmake" => :build
+  depends_on "zlib"
```
Comment (agramfort) [inline openmeeg.rb]: you need libmatio too

File: openmeeg.rb [hunk 2]
```diff
@@ -1,15 +1,28 @@
 class Openmeeg < Formula
-  homepage "http://www-sop.inria.fr/athena/software/OpenMEEG/"
-  url "https://github.com/openmeeg/openmeeg/archive/release-2.1.tar.gz"
-  sha256 "fa424c59a4366d6331c76e539cf5df21bf5e0b065972b2bdda20cdb9f29aba6a"
+  desc "Package for low-frequency bio-electromagnetism solving"
```
Comment (agramfort) [inline openmeeg.rb]: Package for low-frequency bio-electromagnetism forward models

File: openmeeg.rb [hunk 3]
```diff
@@ -1,15 +1,28 @@
...
+      -DBUILD_SHARED_LIBS=ON
+      -DENABLE_PYTHON=OFF
+      -DBUILD_DOCUMENTATION=OFF
+      -DUSE_PROGRESSBAR=ON
+    ]
+
+    mkdir "build" do
+      args << "../OpenMEEG"
+      system "cmake", *args
+      system "make", "install"
```
Comment (agramfort) [inline openmeeg.rb]: to test you can use

make test

**Comment by agramfort** [comment]
see 

https://github.com/openmeeg/openmeeg/blob/master/build_tools/travis/before_install.sh

we do:
        brew install hdf5
        brew install libmatio --with-hdf5

**Comment by iMichka** [comment]
>  * Dependency libmatio should not use option with-hdf5

PR opened here: #5771

**Comment by iMichka** [comment]
@agramfort For the tests, I am thinking about something simpler than `make test`. `make test` is more a tool for you. Some test suites run for a very long time. Basically what we often use if a simple example code that needs to be compiled; or if the library is a command line too, we just invoke it and check the return code. The constraint is that when we run the tests the build dir is gone, so we only rely on what is installed. It is more a sort of "runtime" check.

**Comment by agramfort** [comment]
ok then run the om_assemble command. It should show the
help and return without error.

**Comment by iMichka** [comment]
@agramfort The formula is now correctly setup, and ready to be pushed. I would like to know if you would maybe consider tagging this version with a release tag (2.4 in that case?). If not we will distribute this as 2.4.prerelease the time being.

**Comment by agramfort** [comment]
We'll tag 2.4 over the next two weeks.

I'll send you a PR here to update the tag when it's ready.

Feel free to merge.

Thanks so much of your help.

cc @eolivi @papadop

**Comment by agramfort** [comment]
@iMichka would you have time to help us shipping openmeeg with an option to use openmp by compiling with clang-omp ?

basically it's changing the compiler with cmake and use USE_OMP=ON option.

we use it on travis with:

https://github.com/openmeeg/openmeeg/blob/master/build_tools/travis/before_install.sh#L31

I have in mind something like this:

brew install openblas --with-omp

would this be consistent with other packages?

thanks for any input

cc @eolivi @papadop

**Comment by ilovezfs** [comment]
Just a heads up that iMichka is on vacation for the next couple weeks

**Comment by agramfort** [comment]
ok no rush. thanks

**Comment by agramfort** [comment]
@iMichka if you're back from vacation do you mind having a look at my message above about openmp support ? thanks a lot

**Comment by iMichka** [comment]
I'll be back next week. Did you already try to edit your local formula and build it to see if it works fine? I can help to write the PR but I would prefer you submit a first version sonI can review it.

**Comment by agramfort** [comment]
have a look at #6022

**Pull request closed without merging**